clients/client/src/main/java/org/projectnessie/client/ClientTreeApi.java

```diff
@@ -106,10 +106,15 @@ public Branch getDefaultBranch() {
 
   @Override
   public LogResponse getCommitLog(@NotNull String ref, @Nullable Integer maxEntriesHint,
-      @Nullable String pageToken) throws NessieNotFoundException {
+      @Nullable String pageToken, @Nullable String author, @Nullable String committer,
```

Comment:
can we use an object here rather than a `method(String, String, String, String, String)`? I thought there was somethign in JAX-RS that would even convert the object into query params. I think the method signature is real easy to get messed up.

Also do we want to support these as Lists? to fetch all commits by eg author1,author2,author3

Reply:
I would actually more inclined to use some query language (something like https://opensource.google/projects/cel) in order to give some flexibility in term of querying

Reply:
unfortunately there's no Java implementation for CEL :(

Reply:
I think we want to follow the git semantics as much as possible here. So looking at `git log --help` we should be able to do:

* before a date
* after a date
* multiple authors/committers and regex support for author/committer

We will eventually want to add regex to filter on log messages, file paths etc so my thinking is we need something more than adding `String` parameters to the function signature. Whether we need a full fledged query language for the query params at this stage is up for debate but if there is an easy to use lib it would be nice.

Reply:
nit: before and after should likely be date/instant etc objects

Reply:
I agree having X parameters on a method is super annoying. I'll switch stuff to using `@BeanParam` and put all parameters into a Wrapper class.

> Also do we want to support these as Lists? to fetch all commits by eg author1,author2,author3

Not sure, maybe as a follow-up after this PR?

Reply:
> Not sure, maybe as a follow-up after this PR?

don't see a reason to not do it now.

Reply:
added support for filtering by multiple authors/committers